src/Nox.Lib/Presentation/Api/OData/ApiRouteQueryStringOperationFilter.cs

```diff
@@ -0,0 +1,37 @@
+using Microsoft.OpenApi.Models;
+using Nox.Solution;
+using Swashbuckle.AspNetCore.SwaggerGen;
+
+namespace Nox.Presentation.Api.OData;
+
+public class ApiRouteQueryStringOperationFilter: IOperationFilter
+{
+    private readonly IEnumerable<string> _auditEntityPaths;
+
+    public ApiRouteQueryStringOperationFilter(NoxSolution noxSolution)
+    {
+        _auditEntityPaths = 
+            noxSolution.Domain!.Entities.Where(e => e.Persistence.IsAudited)
+                .Select(e=> string.Concat(noxSolution.Presentation.ApiConfiguration.ApiRoutePrefix.Trim('/'), "/", e.PluralName));
+    }
+    
+    public void Apply(OpenApiOperation operation, OperationFilterContext context)
+    {
+        if (context.ApiDescription.HttpMethod == HttpMethod.Get.Method && _auditEntityPaths.Any(p => context.ApiDescription.RelativePath!.Contains(p)))
+        {
+            operation.Parameters.Add(new OpenApiParameter
+            {
+                Name = "lang",
+                Description = "language parameter",
+                In = ParameterLocation.Query,
+                Style = ParameterStyle.Simple,
+                Schema = new OpenApiSchema
+                {
+                    Type = "string",
+                    Pattern = "^[a-z]{2}-[A-Z]{2}$"
```

Comment:
see the Nox.Type culture code regular expressions. And in Nox.Solution.Localizaton you can see a unique regular expression that merges all of them

Reply:
Desired changes have been applied